server/controllers/users/addUser/index.js

```diff
@@ -0,0 +1,50 @@
+const { hash } = require('bcrypt');
+
+const { addUserSchema, boomify } = require('../../../utils');
+const {
+  checkUserIdByEmail,
+  addUserQuery,
+} = require('../../../database/queries');
+const { sign } = require('../../../utils/jwt');
+
+const addUser = async (req, res, next) => {
+  try {
+    const { email, password } = await addUserSchema.validate(req.body, {
+      abortEarly: false,
+    });
+
+    const {
+      rows: [userByEmail],
+    } = await checkUserIdByEmail({ email });
+
+    if (userByEmail) {
+      return next(boomify(409, 'checking email', 'This email already exists'));
+    }
+
+    const hashedPassword = await hash(password, 10);
```

Comment:
you can use getSalt() method to make salt with an optional log rounds
```suggestion
    const salt = await genSalt(10);
    const hashedPassword = await hash(password,salt);
```

Reply:
the hash function auto-generates the salt for more details check 
link:https://medium.com/@manishsundriyal/a-quick-way-for-hashing-passwords-using-bcrypt-with-nodejs-8464f9785b67